### DISCUSSION TITLE
关于 Table 组件的疑问!

### DISCUSSION THREAD

**Discussion by mingyuantech** [ant-design/ant-design#29469] [General]
table 组件为什么不用一个 table 然后用 position: sticky; 实现, 要用几个 div 和 table 来做呢?

**Comment by yoyo837** [comment]
一个table做不了列冻结+fixed header的所有功能

**Reply by mingyuantech** [reply]
可以实现啊, 这个 MP4 可能需要下载下来才能观看.
https://user-images.githubusercontent.com/11971917/109094282-a4c0f780-7754-11eb-87f5-6e6dde2154f0.mp4

![image](https://user-images.githubusercontent.com/11971917/109094323-b99d8b00-7754-11eb-8672-e73f2ce2fbf2.png)
![image](https://user-images.githubusercontent.com/11971917/109094388-d2a63c00-7754-11eb-9f00-c969bca6531d.png)

**Reply by yoyo837** [reply]
不记得还有没有其他考虑，你可以试试在rc-table里PR改进

**Reply by mingyuantech** [reply]
目前 table 组件要固定列时会固定宽度, 不能做到自适应不是很方便.
我是在封装一个 table 组件, 然后就看看 ant-design 来参考一下, 发现 ant-design 的实现方式是这样的, 所以就好奇问一下.
因为两者的实现方向不同, 如果 ant-design 要修改的话, 应该是需要重构 table 组件的.
